jscomp/core/js_implementation.ml

```diff
@@ -166,7 +166,7 @@ let after_parsing_impl ppf outputprefix (ast : Parsetree.structure) =
         in
         let js_program =
           print_if_pipe ppf Clflags.dump_rawlambda Printlambda.lambda lambda
-          |> Lam_compile_main.compile outputprefix exports
+          |> Lam_compile_main.compile outputprefix NodeJS exports
```

Comment:
Not sure about passing NodeJS as always here.

Reply:
That does not sound right. Unless, this is only used by the code you have added?
Can you explain the context more here?

Reply:
Maybe we can try using `Js_packages_info.t` inside `lam_compile_primitive` directly where the module_system needs to know.

Reply:
> That does not sound right. Unless, this is only used by the code you have added? Can you explain the context more here?

Pimport needs to know about the information what module_system is. That's why the argument drilling was all over the place in lam along with output_prefix. We can leave the output_prefix but module_system.

Reply:
I think it would help to first reduce the number of lines changes. As right now it's really hard to see where this is used.

Reply:
Eg. instead of
let rec foo x y ... and bar x y ... and ...
do

let toplevel x y = 
  let rec foo ... and bar ...
  
  So most of the change becomes only indentation.

Reply:
Yeap, let me try experiments to get module_system from Js_packages_info inside Pimport.

Reply:
> Yeap, let me try experiments to get module_system from Js_packages_info inside Pimport.

Yes that would clean all of this, if it works.

Reply:
Side comment: The previous design did not require a module_system when writing a js_program with lam because it only needed a module_system to represent the import(top of file) and export(bottom of file) of the js output. However, the dynamic import feature requires a module_system in the js_program as well.

Reply:
Side side comment.  I wonder for how long we'll need to support all these module systems. It would be nice to converge to es6 at some point.